pkg/services/store/entity/sqlstash/sql_storage_server.go

```diff
@@ -224,14 +233,27 @@ func readEntity(rows *sql.Rows, r FieldSelectRequest) (*entity.Entity, error) {
 }
 
 func (s *sqlEntityServer) Read(ctx context.Context, r *entity.ReadEntityRequest) (*entity.Entity, error) {
+	ctx, span := s.tracer.Start(ctx, "storage_server.Read")
+	defer span.End()
+	span.SetAttributes(attribute.String("request", r.String()))
+	s.log = s.log.FromContext(ctx)
+
 	if err := s.Init(); err != nil {
+		s.log.Error("init error", "error", err, "method", "read")
 		return nil, err
 	}
 
-	return s.read(ctx, s.sess, r)
+	res, err := s.read(ctx, s.sess, r)
+	if err != nil {
+		s.log.Error("read error", "error", err, "method", "read")
+	}
+	return res, nil
```

Comment:
should this still return the err?
```suggestion
	return res, err
```

Reply:
yes 😓 good catch, thanks!